src/elements/Button/Button.tsx

```diff
@@ -42,6 +42,16 @@ export interface ButtonProps
    * If true, the animation of the button will be disabled.
    */
   disableAnimation?: boolean;
+
+  /**
+   * Icon to display before the Button label.
+   */
+  startIcon?: React.ReactNode;
```

Comment:
I wouldn't call it `icon` because you theortically could put something else here - like a emoji. Let's loosen this type and tweak the name to be more generic like the other useages.

Reply:
Renamed to `startAdornment` / `endAdornment`